### PR TITLE
Dynamically import core component

### PR DESCRIPTION
**Pull request by nawa-diverta**
https://github.com/diverta/Kuroco-opendev/issues/4295

### CODE REVIEW

File: packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue [hunk 1]
```diff
@@ -1,32 +1,49 @@
 <template>
     <div>
```
Comment (yabe-diverta) [inline packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue:2]: I guess no need to wrap `<div>`, just use `<ExtRelationFld v-if="flagToLoadOnce" v-bind="..." />` on root level.

File: packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue [hunk 2]
```diff
@@ -1,32 +1,49 @@
...
-        ExtRelationFld: () => import('./ExtRelationFld/index.vue'),
+    data() {
+        return {
+            flagToLoadOnce: false,
+        };
+    },
+    computed: {
+        isLoaded() {
+            // isLoaded in ContentsGroupingExtension
+            return this.$parent.$parent.isLoaded;
```
Comment (yabe-diverta) [inline packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue:33]: I think it's not good checking parent components.
Please consider just passing prop instead.

File: packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue [hunk 3]
```diff
@@ -1,32 +1,49 @@
...
+        },
+    },
+    watch: {
+        isLoaded: {
+            immediate: true,
+            handler(newVal) {
+                if (newVal && !this.flagToLoadOnce) {
+                    console.log(window['common/components/extensions/ExtRelationFld']);
```
Comment (yabe-diverta) [inline packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue:41]: Will remove

File: packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue [hunk 4]
```diff
@@ -44,12 +44,9 @@ export default {
         ParentDropdown,
         ChildImage: () => import(/* webpackChunkName: "ChildImage" */ '@/components/ChildImage.vue'),
```
Comment (yabe-diverta) [inline packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue:45]: Need to check correctly loads Webpack chunk from global lib.

File: packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue [hunk 5]
```diff
@@ -125,17 +123,50 @@ export default {
         sortByExtOrderNumber(extA, extB) {
             return extB.ext_order_no - extA.ext_order_no;
         },
+        loadScript(src) {
```
Comment (yabe-diverta) [inline packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue:126]: Is this required to use async/await?

File: packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue [hunk 6]
```diff
@@ -125,17 +123,50 @@ export default {
...
+
+        const prefixUrl = '/management/js/rcms-vue/components/rcms-mng/';
+
+        // load manifest.json to get each ext components' file name.
+        const kurocoCoreManifestUrl = prefixUrl + 'manifest.json?v=' + Date.now();
+        const manifest = await fetch(kurocoCoreManifestUrl).then((res) => res.json());
+
+        try {
+            await this.loadScript(prefixUrl + manifest['rcms-mng-vendors.js']);
+            const coreComponents = ['Text', 'Textarea', 'Link', 'RelationFld'];
```
Comment (yabe-diverta) [inline packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue:155]: Will add other ext components after @aayusharyan adds them.

**Review by yabe-diverta — CHANGES_REQUESTED** [review]
Have to check either this component works fine on local at first.
And do some tasks next.

Suspend until test ready.
- [ ] @takel-shiba please introduce me how to test this on local.
- [ ] remove all child components in this dir.
- [ ] check to load child components asynchronously (to reduce initializing time)

File: packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue (hunk 3 above)
Comment (nawa-diverta) [reply packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue:41]: @yabe-diverta 

Oh yes, sorry, I was debugging it 😅

File: packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue (hunk 5 above)
Comment (nawa-diverta) [reply packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue:126]: @yabe-diverta 

We only need to load the core's manifest and core's vendor js first. 
Child components will be loaded asynchronously afterwards with the following 🤔 
```
await Promise.all(
                coreComponents.map((component) =>
                    this.loadScript(prefixUrl + manifest['common/components/extensions/Ext' + component + '.js'])
                )
            );
```

Sorry, is that what you mean?

**Comment by nawa-diverta** [comment]
Points to consider:
- Since the Vue instance of components are not included in the built source, some dependencies such as `translate` function in Mixin and Vuex are needed to be implemented in the plugin side. This could be improved later by eliminating the dependencies somehow (maybe).
- Since the core component should be module-agnostic, e.g. module Member can use it too, so Topic's related elements like the following should be kept in plugin side.
https://github.com/diverta/management-vue-plugin-sample/blob/18a5be5e84eb450f62fd21302ed24644c10a523f/packages/ContentsGroupingExtensions/src/components/ChildText.vue#L3-L9

File: packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue (hunk 4 above)
Comment (yabe-diverta) [reply packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue]: loaded expectedly

File: packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue (hunk 3 above)
Comment (yabe-diverta) [reply packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue]: removed

File: packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue (hunk 2 above)
Comment (yabe-diverta) [reply packages/ContentsGroupingExtensions/src/components/ChildRelationField/index.vue]: fixed

File: packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue (hunk 6 above)
Comment (yabe-diverta) [reply packages/ContentsGroupingExtensions/src/pages/ContentsGroupingExtension.vue]: added

**Review by yabe-diverta — APPROVED** [review]
No issues found for me, Approved.

@aayusharyan @nawa-diverta 
However I just want to ask a point:
ckfinder script tags are there in header multiplied,
but won't fetch 2 or more in Network tab.
Is this expected?
(Is the ckfinder loaded only one time?)

**Comment by nawa-diverta** [comment]
@yabe-diverta @aayusharyan 

Thank you for the review!

> However I just want to ask a point:
ckfinder script tags are there in header multiplied,
but won't fetch 2 or more in Network tab.
Is this expected?
(Is the ckfinder loaded only one time?)

Hmm not so sure yet, it might be browser's feature... 🤔

**Comment by nawa-diverta** [comment]
@yabe-diverta 

BTW I have uploaded this PR built source plugin to dev-rkb 🙇

**Pull request merged**